features/proposed/FxA-76-metrics-opt-out/README.md

```diff
@@ -0,0 +1,194 @@
+# Metrics Opt-Out
+
+https://mozilla.aha.io/features/FXA-76
+
+## Stories
+
+As a privacy conscious user,
+I want the ability to opt-out of non-essential metrics gathering
+from Firefox Accounts and related services.
+
+As a user, I want Firefox Accounts to
+automatically respect the data choices
+that I've already expressed in my local Firefox settings.
+
+
+## Details
+
+Firefox users can make choices about the data they share with Mozilla
+by using the following UI in Firefox:
+
+![](browser_data_choices.png)
+
+Which toggle the following options in about:config:
+
+* datareporting.healthreport.uploadEnabled
+* toolkit.telemetry.enabled
+
+Which in turn control whether
+the browser will collected and submit
+usage and performance metrics.
+
+Firefox Accounts should provide a similar level of control
+over its metrics data collection,
+and should automatically respect the user's data choices
+that they've already made in the browser.
+We'll do this by adding
+a single opt-out preference
+to control metrics gathering behaviour.
+
+
+### Metrics opt-out preference
+
+We will add an additional field
+called "metrics\_optout"
+to the user's profile data,
+which will have three possible states:
+
+* `null`: the user has not made an explicit choice to disable metrics.
+* `false`: the user has explicitly chosen to allow metrics collection.
+* `true`: the user has explicitly chosen to disable metrics collection.
+
+The default will be `null`,
+and metrics-gathering will be assumed enabled by default,
+analogous to the way Health Report
+is enabled by default in Firefox.
+
+The preference will be managed by fxa-profile-server
+in the same manner as existing profile data elements,
+using the following new API endpoints:
+
+* `GET /v1/metrics_optout`
+* `POST /v1/metrics_optout`
+
+The preference will be readable by any relier
+granted the "profile:metrics\_optout" scope,
+which will only be available
+to trusted internal reliers.
+Mozilla properties that rely on Firefox Accounts
+should consider respecting the user's preference,
+although we will not make it a hard requirement
+for becoming a relier.
+
+### Effects of the preference
+
+For users with `metrics_optout=true`,
+we will disable non-essential metrics collection by:
+
+* Disabling submission of in-page metrics from the content-server
```

Comment:
Good question.  Perhaps it would suffice to message the value of `metrics_optout` up to the firstrun page and let them make their own decision.